test/system/system_test.go

```diff
@@ -137,6 +137,40 @@ var _ = Describe("Kratix", func() {
 				})
 			})
 
+			PWhen("an existing resource request is updated", func() {
+				const requestName = "update-test"
+				var oldNamespaceName string
+				BeforeEach(func() {
+					oldNamespaceName = fmt.Sprintf("old-%s", requestName)
+					createNamespace := fmt.Sprintf(
+						`kubectl create namespace %s --dry-run=client -oyaml > /kratix/output/namespace.yaml`,
+						oldNamespaceName,
+					)
+					platform.kubectl("apply", "-f", requestWithNameAndCommand(requestName, createNamespace))
+					platform.kubectl("wait", "--for=condition=PipelineCompleted", "bash", requestName, pipelineTimeout)
+					worker.eventuallyKubectl("get", "namespace", oldNamespaceName)
+				})
+
+				It("executes the update lifecycle", func() {
+					newNamespaceName := fmt.Sprintf("new-%s", requestName)
+					updateNamespace := fmt.Sprintf(
+						`kubectl create namespace %s --dry-run=client -oyaml > /kratix/output/namespace.yaml`,
+						newNamespaceName,
+					)
+					platform.kubectl("apply", "-f", requestWithNameAndCommand(requestName, updateNamespace))
+
+					By("redeploying the contents of /kratix/output to the worker destination", func() {
+						Eventually(func() string {
+							return worker.kubectl("get", "namespace")
+						}, "10s").Should(
+							SatisfyAll(
+								Not(ContainSubstring(oldNamespaceName)),
+								ContainSubstring(newNamespaceName),
+							),
+						)
+					})
```

Comment:
worth adding that the old namespace is deleted as well?

Reply:
thats line 167?

Reply:
![image](https://github.com/syntasso/kratix/assets/19669095/49e0d3e3-68cc-4a2a-bb86-a49f09e878de)